nixos/modules/services/databases/postgresql.nix

```diff
@@ -192,6 +193,17 @@ in
         '';
       };
 
+      logLinePrefix = mkOption {
+        type = types.str;
+        default = "[%p] ";
```

Comment:
Shouldn't we remove the `%p` as well then? The PID is also logged in `journald`.

Reply:
Journal only shows (and records, IIUC) the PID of the main process, while postgres spawns and logs from a lot of subprocesses.

To be honest, I am not exactly sure if this is how it is supposed to work, i.e. why journalctl always shows the PID of the parent process – I would expect it to save the PID of the process that _actually_ wrote the message. This might be something worth looking into later.

Reply:
Uhm, AFAICS on my machine, journald doesn't show the parent's PID on each log-line, I'm using systemd 245 though.

Reply:
I don’t know about that. [The code](https://github.com/systemd/systemd/blob/7d85383edbab73274dc81cc888d884bb01070bc2/src/shared/logs-show.c#L477) that prints it has been there since, at least, 2012. And if you google “systemd” and go to images, you’ll see that they all contain the pid.

Reply:
not sure if I'm misunderstanding you, but I meant that the PID of the process producing the line is shown in the line and not the PID of the main process of a unit.

Reply:
Oh! Sorry, I misunderstood you.

I don’t see any release notes in the last couple versions that seem to be relevant. Can you try specifically with postgres please? My theory is that maybe this wrapping into bash scripts that we do breaks something. What I see in my journal is that the PID that `journalctl` prints is always the same for `postgresql.service` and the PIDs printed by postgres itself are different.